modules/ipfs-cpinner-client/src/encryption-manager/with-signer.ts

```diff
@@ -0,0 +1,31 @@
+import { IEncryptionManager } from './types'
+import { encrypt, decrypt, generateKeyViaRPC } from './aes'
+import { Web3Provider } from '../web3provider/types'
+
+class EncryptionManager implements IEncryptionManager {
+  private key: string
+  private macKey: string
+
+  constructor (key: string, macKey: string) {
+    this.key = key
+    this.macKey = macKey
+  }
+
+  encrypt = (data: string) => {
+    const cipher = encrypt(this.key, data, this.macKey)
+    return Promise.resolve(Buffer.from(cipher, 'utf-8').toString('hex'))
+  }
+
+  decrypt = (data: string) => {
+    const cipher = Buffer.from(data, 'hex').toString('utf-8')
+    return Promise.resolve(decrypt(this.key, cipher, this.macKey))
+  }
```

Comment:
data should already be an hex string so no need to convert it again

```suggestion
  decrypt = (cipher: string) => {
    return Promise.resolve(decrypt(this.key, cipher, this.macKey))
  }
```